### PR TITLE
Unit test for status.address.hostname

### LINKED ISSUE #475: GitHubSource doesn't work with Serving 0.7
## Expected Behavior
I can target a Serving 0.7 Service with a GitHubSource.

## Actual Behavior
Doesn't work because Service 0.7 doesn't populate the `status.address.hostname` field.

## Additional Info
Probably an issue for all sources.
See https://knative.slack.com/archives/C9JP909F0/p1561666023287200

### PR DESCRIPTION
**Pull request by Harwayne**
## Proposed Changes

  * Add a unit test for #476.

**Release Note**

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/sinks/sinks_test.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ import (
 
 var (
 	addressableDNS = "addressable.sink.svc.cluster.local"
+	addressableURL = fmt.Sprintf("http://%s/", addressableDNS)
```
Comment (matzew) [inline pkg/controller/sinks/sinks_test.go]: In #477 I was doing a _straight_ port of what @mattmoor did,  so, only updating the 

```
			want:      fmt.Sprintf("http://%s", addressableDNS),
```

Note, that the format here is different `%s` versus `%s/` (trailing `/`) on your PR

**Comment by matzew** [comment]
generally wondering, what best to avoid some of this code duplications, here and eventing (core)

**Comment by matzew** [comment]
@Harwayne in 477, I had to (similar to Matt) also fix some of the URI expectations, on the tests

**Comment by n3wscott** [comment]
In v1beta1 GetURL goes away.

File: pkg/controller/sinks/sinks_test.go (hunk 1 above)
Comment (grantr) [reply pkg/controller/sinks/sinks_test.go]: Agree with @matzew, we should probably not add a trailing slash here. Golang's `url.URL.String()` doesn't: https://play.golang.org/p/cFaC63TxyXd

File: pkg/controller/sinks/sinks_test.go [hunk 2]
```diff
@@ -63,7 +64,15 @@ func TestGetSinkURI(t *testing.T) {
 		wantErr   error
 		ref       *corev1.ObjectReference
 	}{
-		"happy": {
+		"happy - hostname": {
```
Comment (grantr) [inline pkg/controller/sinks/sinks_test.go]: I like the addition of this test ensuring we can still target v1alpha1 Addressables.

File: pkg/controller/sinks/sinks_test.go (hunk 1 above)
Comment (Harwayne) [reply pkg/controller/sinks/sinks_test.go]: Removed.

**Pull request merged**